### PR TITLE
Better server tests

### PR DESCRIPTION
**Pull request by freiksenet**
Depends on #40

### CODE REVIEW

**Comment by stubailo** [comment]
@freiksenet just rebased this, now learning how to run tests locally

**Comment by stubailo** [comment]
Hmmmm...

```
Summary of all failing tests
 FAIL  frontend/__tests__/Editor.test.js
  ● renders

    TypeError: (0 , _EditorLinter2.default) is not a function

      at Editor.componentDidMount (frontend/Editor.js:31:32)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:265:25
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:75:12)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:264:11
      at CallbackQueue.notifyAll (node_modules/react-dom/lib/CallbackQueue.js:76:22)
      at ReactReconcileTransaction.close (node_modules/react-dom/lib/ReactReconcileTransaction.js:80:26)
      at ReactReconcileTransaction.closeAll (node_modules/react-dom/lib/Transaction.js:206:25)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:153:16)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:126:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:140:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:62:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:97:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:320:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:401:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:422:23)
      at Object.renderIntoDocument (node_modules/react-dom/lib/ReactTestUtils.js:91:21)
      at renderWithOptions (node_modules/enzyme/build/react-compat.js:209:26)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:94:59)
      at mount (node_modules/enzyme/build/mount.js:19:10)
      at Object.<anonymous>.test (frontend/__tests__/Editor.test.js:17:39)
          at Promise (<anonymous>)
      at Promise.resolve.then.el (node_modules/p-map/index.js:42:16)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:169:7)


Test Suites: 1 failed, 20 passed, 21 total
Tests:       1 failed, 63 passed, 64 total
Snapshots:   54 passed, 54 total
Time:        6.538s
```

going to review the tests now since they seem to run!

File: server/PadModel.js [hunk 1]
```diff
@@ -37,7 +37,10 @@ const PadModel = {
     }
 
     let draft = null;
-    if (pad.draft && UserModel.canUpdatePadDraft(user, pad.draft, context)) {
+    if (
+      pad.draft &&
+      (!pad.user || UserModel.isPadOwner(user, pad.draft, context))
```
Comment (stubailo) [inline server/PadModel.js]: This is duplicated from line 29, looks like we need a `UserModel.canSeePadSecrets`?

File: server/PadModel.js [hunk 2]
```diff
@@ -325,7 +328,7 @@ const PadModel = {
     let pad = await PadModel.getById(id, context);
 
     const user = UserModel.me(context);
-    if (!UserModel.canUpdatePadDraft(user, pad, context)) {
+    if (pad && pad.user && !UserModel.isPadOwner(user, pad, context)) {
```
Comment (stubailo) [inline server/PadModel.js]: Hmm, couldn't this just be `UserModel.isPadOwner` and then that function can handle `pad` or `pad.user` being absent?

**Comment by stubailo** [comment]
Not sure what that frontend test error is about.

File: server/PadModel.js (hunk 1 above)
Comment (freiksenet) [inline server/PadModel.js]: You are right about that, we should split the permissions by type of action.

File: server/PadModel.js (hunk 2 above)
Comment (freiksenet) [inline server/PadModel.js]: Yeah, I think we should clearly split those functions.

**Comment by freiksenet** [comment]
@stubailo Clarified permissions. I don't get frontend error neither does CI, are you using npm 5? Sounds like dep problem.

**Pull request merged**